File: app/src/main/java/by/offvanhooijdonk/plaincalendar/widget/data/calendars/observer/EventsContentObserver.java
package by.offvanhooijdonk.plaincalendar.widget.data.calendars.observer;

import android.database.ContentObserver;
import android.os.Handler;
import android.util.Log;

import static by.offvanhooijdonk.plaincalendar.widget.app.App.LOGCAT;

public class EventsContentObserver extends ContentObserver {
    private EventsChangeListener listener;

    public EventsContentObserver(Handler handler, EventsChangeListener l) {
        super(handler);

        this.listener = l;

        Log.i(LOGCAT, "Content Observer created");
    }

    @Override
    public void onChange(boolean selfChange) {
        super.onChange(selfChange);
        Log.i(LOGCAT, "onChange");

        if (listener != null) {
            listener.onDataChanged();
        }
    }

    @Override
    public boolean deliverSelfNotifications() {
        return true;
    }

    public interface EventsChangeListener {
        void onDataChanged();
    }
}

File: app/src/main/java/by/offvanhooijdonk/plaincalendar/widget/data/calendars/observer/EventsContentObserver.kt
package by.offvanhooijdonk.plaincalendar.widget.data.calendars.observer

import android.database.ContentObserver
import android.os.Handler
import android.util.Log
import by.offvanhooijdonk.plaincalendarv2.widget.app.App

class EventsContentObserver(handler: Handler, private val onDataChanged: () -> Unit) : ContentObserver(handler) {

    init {
        Log.i(App.LOGCAT, "Content Observer created")
    }

    override fun onChange(selfChange: Boolean) {
        super.onChange(selfChange)
        Log.d(App.LOGCAT, "onChange")
        onDataChanged()
    }

    override fun deliverSelfNotifications(): Boolean {
        return true
    }
}
